src/player.cpp

```diff
@@ -1824,6 +1824,16 @@ bool player::is_immune_effect( const efftype_id &effect ) const
     return false;
 }
 
+bool player::drag_check() const
+{
+    if (get_str() + (get_per() / 3) + (get_dex() / 4) > rng(4, 15) || is_throw_immune()){
```

Comment:
You could use something like in `smash`: just a formula of monster's attack dice.